src/accel/LocalTransporter.cc

```diff
@@ -57,6 +62,47 @@ bool nonfatal_flush()
     }();
     return result;
 }
+#if CELERITAS_USE_PERFETTO
+//! Trace the number of active, alive, dead, and queued tracks
+class TrackCounters
+{
+  public:
+    TrackCounters()
+    {
+        if (use_profiling())
+        {
+            std::string stream_id = std::to_string(get_geant_thread_id());
+            active_counter_ = std::string("active-" + stream_id);
+            alive_counter_ = std::string("alive-" + stream_id);
+            dead_counter_ = std::string("dead-" + stream_id);
+            queued_counter_ = std::string("queued-" + stream_id);
+        }
+    };
+
+    void operator()(StepperResult const& track_counts)
+    {
+        trace_counter(active_counter_.c_str(), track_counts.active);
+        trace_counter(alive_counter_.c_str(), track_counts.alive);
+        trace_counter(dead_counter_.c_str(),
+                      track_counts.active - track_counts.alive);
+        trace_counter(queued_counter_.c_str(), track_counts.queued);
+    }
+
+  private:
+    std::string active_counter_;
+    std::string alive_counter_;
+    std::string dead_counter_;
+    std::string queued_counter_;
+};
+
+void trace(StepperResult const& track_counts)
+{
+    static thread_local TrackCounters trace_;
+    trace_(track_counts);
+}
+#else
+void trace(StepperResult const&) {}
+#endif
```

Comment:
Since we don't have any perfetto-enabled CI builds, and `trace_counter` is a null-op, I'd rather have this class be "always-on" so that it's guaranteed to compile. Hopefully the compiler will optimize it (aside from the negligible four static std::string storage) out if perfetto is disabled.

Reply:
Since `trace_counter` depends on the runtime environment, it's not always possible to optimize it to null-op, but the cost is very small, one `if(false)`